src/piral-build.test.ts

```diff
@@ -7,63 +7,180 @@ runTests('piral-build', ({ test, setup }) => {
     await ctx.run(`npm i emojis-list@3.0.0`);
   });
 
+  test('release-standard-template', 'can produce a release build', ['build.piral'], async (ctx) => {
+    await ctx.run(`npx piral build --type release`);
+
+    await ctx.assertFiles({
+      'dist/release/index.html'(content: string) {
+        expect(content).not.toBe('');
+      },
+      'dist/emulator': false,
+    });
+  });
+
+  test('emulator-standard-template', 'can produce an emulator build', ['build.piral'], async (ctx) => {
+    await ctx.setFiles({
+      'package.json'(content: string) {
+        const packageJson = JSON.parse(content);
+        packageJson.name = 'app-shell';
+        return JSON.stringify(packageJson, undefined, 2);
+      },
+    });
+
+    await ctx.run(`npx piral build --type emulator`);
+
+    await ctx.assertFiles({
+      'dist/emulator/*.tgz'(files: Array<string>) {
+        expect(files).toHaveLength(1);
+        ctx.setRef('emulator', files[0]);
+      },
+      'dist/release': false,
+    });
+  });
+
+  test('emulator-sources-standard-template', 'can produce emulator sources build', ['build.piral'], async (ctx) => {
+    await ctx.run(`npx piral build --type emulator-sources`);
+
+    await ctx.assertFiles({
+      'node_modules/piral-cli/package.json': true,
+      'dist/emulator/package.json': true,
+      'dist/emulator/app/index.html': true,
+      'dist/release': false,
+    });
+  });
+
+  test(
+    'some-invalid-type-standard-template',
+    'can not produce build with invalid type',
+    ['build.piral'],
+    async (ctx) => {
+      try {
+        await ctx.run(`npx piral build --type some-invalid-type-standard-template`);
+        expect(true).toBe(false);
+      } catch {}
+
+      await ctx.assertFiles({
+        'dist/release': false,
+        'dist/emulator': false,
+      });
+    },
+  );
+
   test(
-    'release-standard-template',
-    'can produce a release build',
+    'release-standard-template-with-default-public-url',
+    'can produce a release build with default public url',
     ['build.piral'],
     async (ctx) => {
-      await ctx.run(`npx piral build --type release`);
+      await ctx.run(`npx piral build --type release --public-url '/'`);
 
       await ctx.assertFiles({
         'dist/release/index.html'(content: string) {
           expect(content).not.toBe('');
+          expect(content).toContain('<script defer="defer" src="/index.');
```

Comment:
Don't use the full version with `defer="defer"` -> make a check that is independent of `index` and the specific usage of the `script` tag.